.dir-locals.el

```diff
@@ -0,0 +1,28 @@
+;; This causes emacs to abide by formatting rules: No tab characters, except
```

Comment:
This just applies styles similar to clang-format within emacs to files of certain modes in all subdirectories. But mainly, it prevents tabs and trailing whitespace on lines.

Reply:
This is fine if it helps emacs users and does not harm any other user

Reply:
Yes, it only affects emacs users. Any .dir-locals.el in a directory is loaded whenever emacs edits a file in or below that directory.